rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -128,9 +112,10 @@ rcl_logging_ret_t rcl_logging_external_initialize(
       return RCL_LOGGING_RET_ERROR;
     }
 
-    print_ret = rcutils_snprintf(
+    char name_buffer[4096] = {0};
+    int print_ret = rcutils_snprintf(
       name_buffer, sizeof(name_buffer),
-      "%s/.ros/log/%s_%i_%" PRId64 ".log", homedir,
+      "%s/%s_%i_%" PRId64 ".log", logdir,
```

Comment:
`logdir` is being leaked

Reply:
fixed in b50d081c9d24378b47b447969fadee97da2997ba